.github/workflows/macos.yml

```diff
@@ -48,30 +48,39 @@ jobs:
 
       - name: Checking out repository.
         uses: actions/checkout@v2
+        with:
+          submodules: recursive
 
       ## macOS-specific steps
 
+      - name: Pin Xcode version
+        run: sudo xcode-select -s "/Applications/Xcode_12.1.1.app"
+
       - name: Installing dependencies.
         run: |
-          brew install qt jack p7zip
-          brew link --force qt
+          brew update
+          # JACK implies brew link python-3.9, fails due to shipped python binaries
+          brew unlink python@3.9
+          brew link --overwrite python@3.9
+          brew upgrade --force python@3.9
+          brew install qt5 pkg-config jack p7zip
```

Comment:
`jack` has a dependency on `python@3.9`. When it tries to install `python-3.9.1_1` on GHA and link its binaries into the system's common bin path, it finds a symlink/binary(?) in /usr/local/bin that belongs to macOS' shipped python2 installation and errors out. The suggested options to instead overwrite it assume that there was an attempt to install just `python@3.9` and I couldn't get it to work for this dependency use-case.

```
==> Pouring python@3.9-3.9.1_1.catalina.bottle.tar.gz
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/2to3
Target /usr/local/bin/2to3
already exists. You may want to remove it:
  rm '/usr/local/bin/2to3'

To force the link and overwrite all conflicting files:
  brew link --overwrite python@3.9

To list all files that would be deleted:
  brew link --overwrite --dry-run python@3.9

Possible conflicting files are:
/usr/local/bin/2to3 -> /Library/Frameworks/Python.framework/Versions/2.7/bin/2to3
```

Unlinking the pre-installed `python@3.9` package and relinking it somehow, magically, fixes this. The `brew upgrade` is just to make sure that it's not python that's failing anymore.